File: KdiffPairsInArray.cpp
//time complexity: O(n)
//space complexity:O(n)
//approach: using hashmap.
//solved on leetcode? yes
//problem faced? yes, initially got confused but solved it later

class Solution {
public:
    int findPairs(vector<int>& nums, int k) {
        if(nums.size()==0)
            return 0;
        int count=0;
        map<int,int>map;
        for(auto el:nums)
        {
            map[el]++;
        }
        for(auto x:map)
        {
            int val=x.first+k;
            if(map.find(val)!=map.end())
            {
                if(val==x.first && x.second>1)
                    count++;
                else if(val!=x.first && x.second>=1)
                    count++;
            }
        }
        return count;
    }
};

File: KDiffPairsInArray.cs
// Time Complexity :
// Space Complexity :
// Did this code successfully run on Leetcode :
// Any problem you faced while coding this :


// Your code here along with comments explaining your approach

//[3,1,4,1,5] k = 2
 public int FindPairs(int[] nums, int k) {

    int result = 0; 
    if(nums == null || nums.Length == 0)
        return result;

    HashSet<int> set = new HashSet<int>();
    
    for(int i = 0; i < nums.Length; i++)
    {
        set.Add(nums[i]);
    }

    for(int i = 0; i < nums.Length; i++)
    {
        int diff = Math.Abs(k - nums[i]);
        if(set.Contains(dif))
            result++;
    }
    return result;
 }